Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -2627,8 +2627,129 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(Array(buffer.readableBytesView), Array(repeating: 2, count: 100))
         XCTAssertNotEqual(buffer, copy)
     }
+
+    func testWriteOptionalWorksForNilCase() {
+        self.buf.writeString("hello")
+        var startingWithNil: ByteBuffer? = nil
+        startingWithNil.writeBuffer(&self.buf)
+        self.buf.writeString("hello")
+        XCTAssertEqual(self.buf, startingWithNil)
+    }
+
+    func testWriteOptionalWorksForNonNilCase() {
+        self.buf.writeString("hello")
+        var nonNilBuffer: ByteBuffer? = self.buf
+        nonNilBuffer.writeBuffer(&self.buf)
+        self.buf.writeString("hellohello")
+        XCTAssertEqual(self.buf, nonNilBuffer)
+    }
```

Comment:
These two tests don't check the return values of the `write` methods, which is probably why the earlier bug in those return values slipped through.

Reply:
Great catch, thank you!